src/app/tool-editor/object-inspector/common-sections/symbols-section/symbols.component.ts

```diff
@@ -13,112 +13,45 @@ require("./symbols.component.scss");
     template: `
         <div class="form-group" *ngIf="symbolsForm">
         <form>
-               <label class="form-control-label">Symbols</label>
-                      
-               <ol class="list-unstyled">
-                    <li *ngFor="let symbol of symbolsFormList; let i = index;" 
-                        class="removable-form-control symbol-list-item">
-                    
-                        <div *ngIf="!!symbolsForm.controls[symbol.id]">
-                            <input required class="form-control symbol-input"
-                                  type="text"
-                                  [ngClass]="{'invalid-input': !symbolsForm.controls[symbol.id].valid }"
-                                  [formControl]="symbolsForm.controls[symbol.id]">
-        
-                            <div class="remove-icon clickable" 
-                                *ngIf="symbolsFormList.length > 1"
-                                (click)="removeSymbol(symbol)">
-                                <i class="fa fa-trash"></i>
-                            </div>
-                        </div>
-                    </li> 
-                </ol>
-                      
-                <button type="button" 
-                        class="btn btn-link add-btn-link no-underline-hover" 
-                        (click)="addSymbol()">
-                    <i class="fa fa-plus"></i>Add Symbol
-                </button>
+               <label>Symbols</label>
+               <compact-list *ngIf="symbolsForm" 
+                            [addKeyCode]="13"
+                            [formControl]="symbolsForm">
+                </compact-list>
+                     
         </form>
         </div>
 `
 })
 export class SymbolsComponent extends ComponentBase implements ControlValueAccessor {
```

Comment:
SymbolsComponent is incorrectly implementing ControlValueAccessor. Methods writeValue, registerOnChange and registerOnTouch cannot be private.